src/core/lib/security/credentials/tls/grpc_tls_crl_provider.h

```diff
@@ -77,6 +79,46 @@ class CertificateInfoImpl : public CertificateInfo {
   const std::string issuer_;
 };
 
+class DirectoryReloaderCrlProvider : public CrlProvider {
+ public:
+  static absl::StatusOr<std::shared_ptr<CrlProvider>>
+  CreateDirectoryReloaderProvider(
+      absl::string_view directory, std::chrono::seconds refresh_duration,
+      std::function<void(absl::Status)> reload_error_callback);
+  // ~DirectoryReloaderCrlProvider() override;
+};
+
+// Defining this here lets us hide implementation details (and includes) from
+// the header in include
+class DirectoryReloaderCrlProviderImpl
+    : public DirectoryReloaderCrlProvider,
+      public std::enable_shared_from_this<DirectoryReloaderCrlProviderImpl> {
+ public:
+  DirectoryReloaderCrlProviderImpl(::absl::string_view directory,
+                                   ::std::chrono::seconds duration,
+                                   ::std::function<void(absl::Status)> callback)
+      : crl_directory_(directory),
```

Comment:
Can we move the constructor definition to the .cc file, similar to the other methods?